tests/Http/HttpRequestTest.php

```diff
@@ -122,7 +122,7 @@ public function testFullUrlMethod()
         $this->assertEquals('https://foo.com', $request->fullUrl());
 
         $request = Request::create('https://foo.com', 'GET');
-        $this->assertEquals('https://foo.com?coupon=foo', $request->fullUrlWithQuery(['coupon' => 'foo']));
+        $this->assertEquals('https://foo.com/?coupon=foo', $request->fullUrlWithQuery(['coupon' => 'foo']));
```

Comment:
I think from February onwards the changes have not been synced between these two methods. The individual tests do pass, however there should be a test method to actually check that these two methods behave consistently, since they do the same thing in different ways. Their output should be identical. It is not.

The tests **would not pass**, if there was a check for consistency between the two. 

So unless there is a reason behind what we think is an inconsistency, the tests can be updated to include this check. The methods then can be synced to return the version with a trailing separator to indicate the root resource without relying on the browser adding it on one case and the framework adding it on the other.